src/devices/OpenHardwareMonitor/OpenHardwareMonitor.cs

```diff
@@ -0,0 +1,740 @@
+        /// </summary>
+        public sealed class Sensor : IDisposable
+        {
+            private readonly ManagementObject _instance;
+
+            /// <summary>
+            /// Creates a sensor instance
+            /// </summary>
+            public Sensor(ManagementObject instance, string name, string identifier, string parent, SensorType typeEnum)
+            {
+                _instance = instance;
+                Name = name;
+                Identifier = identifier;
+                Parent = parent;
+                SensorType = typeEnum;
+            }
+
+            /// <summary>
+            /// Name of the sensor
+            /// </summary>
+            public string Name { get; }
+
+            /// <summary>
+            /// Sensor identifier (device path)
+            /// </summary>
+            public string Identifier { get; }
+
+            /// <summary>
+            /// Sensor parent
+            /// </summary>
+            public string Parent { get; }
+
+            /// <summary>
+            /// Kind of sensor
+            /// </summary>
+            public SensorType SensorType { get; }
+
+            /// <summary>
+            /// Job associated with updating this value
+            /// </summary>
+            internal MonitoringJob Job
+            {
+                get;
+                set;
+            }
+
+            /// <summary>
+            /// Attempt to query a value for the sensor
+            /// </summary>
+            /// <param name="value">Returned value</param>
+            /// <returns>True if a value was available</returns>
+            public bool TryGetValue(out IQuantity value)
+            {
+                if (!_typeMap.TryGetValue(SensorType, out var elem))
+                {
+                    value = null;
+                    return false;
+                }
+
+                float newValue = Convert.ToSingle(_instance["Value"]);
+                IQuantity newValueAsUnitInstance = elem.Creator(newValue);
+
+                value = newValueAsUnitInstance;
+                return true;
+            }
+
+            /// <summary>
+            /// Attempt to get a value of the provided type
+            /// </summary>
+            /// <typeparam name="T">The type of the quantity to return</typeparam>
+            /// <param name="value">The returned value</param>
+            /// <returns>True if a value of type T could be retrieved</returns>
+            public bool TryGetValue<T>(out T value)
+                where T : IQuantity
+            {
+                if (!_typeMap.TryGetValue(SensorType, out var elem))
+                {
+                    value = default(T);
+                    return false;
+                }
+
+                if (typeof(T) != elem.Type)
+                {
+                    value = default(T);
+                    return false;
+                }
+
+                float newValue = Convert.ToSingle(_instance["Value"]);
```

Comment:
With this pattern, how do we ensure that the unit is created with the right default unit? Are _instance["Value"] always returning the default international unit? I may have miss something here, I would just like to understand

Reply:
Yes, the values returned are always SI units. Which unit belongs to which sensor type is defined in the global _typeMap array.